### PR TITLE
Update documentation for how to build compiler

### PR DESCRIPTION
**Pull request by robinheghan**
The markdown-based documentation is out of date. This is an attempt to update it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,10 +6,19 @@ There are easier ways to install the compiler than compiling the source, you mig
...
+Since Gren 0.4 the compiler is implemented in two parts. The Gren-portion of the compiler lives in `src`, and once built it acts
+as a frontend to the Haskell-portion of the compiler.
 
-Compiling and installing the project should just be a matter of `cabal install`, after which you should be able to run the `gren` command from your command line.
+The end goal is for the entire compiler to be written in Gren, but for now you need to build both Gren- and Haskell-portions of the compiler.
 
-Read the `CONTRIBUTING.md` file for some helpful commands for working on the compiler itself.
+1. Use `devbox shell` to get a terminal with the required dependencies installed.
+2. Use `./build_dev_bin.sh` to build the Haskell-based compiler. You should now have a `./gren` file in your directory.
+3. Build the Gren compiler with `npm run prepublishOnly`
```
[the remark this answers is not in the document]
Comment (blaix) [reply README.md:19]: Also, this requires that you have `node` and `compiler-node` checked out in `..`

File: ROADMAP.md [hunk 2]
```diff
@@ -8,16 +8,5 @@ Also, keep in mind that this only focuses on the big picture. Major features lik
...
-- December 2024 - Re-evaluating how concurrency works in the language. More specifically, we'll re-evaluate how `Task`, `Cmd`, `Process` and effect modules work.
-
-- June 2025 - Re-evaluate records and custom types.
-
-- December 2025 - Re-evaluate interop: ports and kernel code.
-
-- June 2026 - Compile to Web Assembly instead of JS. This will give us proper integers, big integers, hopefully smaller asset size and possibly enable future optimizations in codegen.
-
-- December 2026 - The language is considered complete. Both the compiler and core packages will commit to maintaining backwards compatability from here on out.
+- June 2025 - Rewrite dependency management in Gren
```
[the remark this answers is not in the document]
Comment (robinheghan) [reply ROADMAP.md:11]: Just figured that Dec 2024 would become public knowledge soon enough.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (blaix) [reply README.md:19]: It worked as long as I had gren-lang installed. Here's the diff that works for me on a fresh setup:

```diff
diff --git a/package.json b/package.json
index 32747985..60b1094e 100644
--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "Compiler for the Gren programming language",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "prepublishOnly": "gren make src/Main.gren --optimize --output=compiler.js",
+    "prepublishOnly": "npx --package=gren-lang@0.4.5 -- gren make src/Main.gren --optimize --output=compiler.js",
     "prettier": "prettier -w \"!**/*.json\" .",
     "prettier:check": "prettier -c \"!**/*.json\" ."
   },
@@ -36,6 +36,7 @@
   },
   "homepage": "https://gren-lang.org",
   "dependencies": {
+    "gren-lang": "^0.4.5",
     "postject": "^1.0.0-alpha.6"
   },
   "devDependencies": {
```

File: README.md [hunk 3]
```diff
@@ -6,10 +6,21 @@ There are easier ways to install the compiler than compiling the source, you mig
...
+The end goal is for the entire compiler to be written in Gren.
 
-Read the `CONTRIBUTING.md` file for some helpful commands for working on the compiler itself.
+To build the compiler:
+
+1. Use `devbox run prepare-deps` to setup the required dependencies. The first time you run this it might take a while.
+2. Build the compiler with `devbox run build`.
+
+You can now execute the compiler with `node ./cli.js` or just `./cli.js`. The Gren-part of the compiler will download a pre-built
+binary of the Haskell-part from Github. If you want to run the compiler with the Haskell-part you've built yourself, set the path
```
Comment (blaix) [inline README.md:23]: > with the Haskell-part you've built yourself

might want to mention that this was built to `./gren`

File: README.md [hunk 4]
```diff
@@ -6,10 +6,21 @@ There are easier ways to install the compiler than compiling the source, you mig
...
-Compiling and installing the project should just be a matter of `cabal install`, after which you should be able to run the `gren` command from your command line.
+The end goal is for the entire compiler to be written in Gren.
 
-Read the `CONTRIBUTING.md` file for some helpful commands for working on the compiler itself.
+To build the compiler:
+
+1. Use `devbox run prepare-deps` to setup the required dependencies. The first time you run this it might take a while.
+2. Build the compiler with `devbox run build`.
+
+You can now execute the compiler with `node ./cli.js` or just `./cli.js`. The Gren-part of the compiler will download a pre-built
```
Comment (blaix) [inline README.md:22]: You'll need to open a `devbox shell` first or else change these to start with `devbox run`

**Review by blaix — COMMENTED** [review]
Love the changes. Everything worked great this time. Couple more comments.

**Pull request merged**